pepsico/app_calc.py

```diff
@@ -0,0 +1,140 @@
+import xarray as xr
+
+
+#This is what we should need for the app
+#For the map:
+#-- read_data for histo and 1 model
+#-- seasonal_data for specific years range
+#-- seasanal_data.mean(dim="T") - seasonal_histo.mean(dim="T")
+#-- unit_conversion
+#-- then maybe some other things to beautify map tbd
+#
+#For the ts:
+#-- read_data for all histo and scenarios
+#-- select X and Y
+#-- seasonal_data
+#-- append or figure out what the graph will need as input to plot histo
+#followed by different scenarios
+#-- unit_conversion
+#-- then maybe some other things to beautify map tbd
+
+
+def read_data(scenario, model, variable, region, unit_convert=False):
+    if region == "US-CA":
+        xslice = slice(-154, -45)
+        yslice = slice(60, 15)
+    elif region == "SAMER":
+        xslice = slice(-86, -34)
+        yslice = slice(16, -60)
+    elif region == "SASIA":
+        xslice = slice(59, 94)
+        yslice = slice(42, 7)
+    elif region == "Thailand":
+        xslice = slice(85, 115)
+        yslice = slice(28, 2)
+    data = xr.open_zarr(
+        f'/Data/data24/ISIMIP3b/InputData/climate/atmosphere/bias-adjusted/global'
+        f'/monthly/{scenario}/{model}/zarr/{variable}'
+    )[variable].sel(X=xslice, Y=yslice)
+    if unit_convert :
+        data = unit_conversion(data)
+    return data
+
+
+def seasonal_data(monthly_data, start_month, end_month, start_year=None, end_year=None):
+
+    #NDJ and DJF are considered part of the year of the 1st month
+    if ((end_year != None) and (start_month > end_month)):
+        end_year = end_year + 1
+    #Reduce data size
+    if start_year != None :
+        start_year = str(start_year)
```

Comment:
`slice` will work fine with numbers. No need to convert to strings.

The one situation where you have to use strings is if you're using dates like `'2024-12-31'`, since that can't be expressed as an integer.

Reply:
turns out it doesn't, I get the following error if I don't convert back to string:

TypeError: cannot do slice indexing on DatetimeIndex with these indexers [1981] of type int